.github/workflows/nix.yaml

```diff
@@ -0,0 +1,165 @@
+        with:
+          fetch-depth: 0
+      - name: Configure
+        id: cfg
+        run: |
+          # Configure
+          echo '::notice triggered by: ${{ github.event_name }}'
+          echo '::notice inputs = ${{ toJSON(fromJSON(inputs.config)) }}'
+          echo "::notice checking registries..."
+          echo "needs-docker-hub=false" >> "$GITHUB_OUTPUT"
+          echo "needs-ecr=false" >> "$GITHUB_OUTPUT"
+          jq -r '.editions | map( ."image-registries"[]) | unique | .[]
+                 | if . == "docker.io" then
+                     "needs-docker-hub=true"
+                   elif . == "ghcr.io" then
+                     "needs-ghcr=true"
+                   elif . == "622024652768.dkr.ecr.eu-west-1.amazonaws.com" then
+                     "needs-ecr=true"
+                   else
+                     empty
+                   end
+                 | @sh' \
+                 <<< '${{ inputs.config }}' >> "${GITHUB_OUTPUT}"
+          echo "::notice checking vast-plugins..."
+          echo "needs-vast-plugins=false" >> "$GITHUB_OUTPUT"
+          if [[ $(comm -12 \
+                 <(jq -r '[(.editions[] | .name)] | sort | .[]' <<< '${{ inputs.config }}') \
+                 <(echo vast-{ce,ee}| tr ' ' '\n' | sort)) \
+          ]]; then
+            echo "needs-vast-plugins=true" >> "$GITHUB_OUTPUT"
+          fi
+          jq '@sh "upload-to-github=\(.editions | any(."copy-package"))"' \
+                   <<< '${{ inputs.config }}' >> "$GITHUB_OUTPUT"
+      - name: Configure ssh-agent
+        if: steps.cfg.outputs.needs-vast-plugins == 'true'
+        uses: webfactory/ssh-agent@v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.VAST_PLUGINS_DEPLOY_KEY }}
+      - name: Login to GitHub Container Registry
+        if: steps.cfg.outputs.needs-ghcr
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: tenzir-bot
+          password: ${{ secrets.TENZIR_BOT_GITHUB_TOKEN }}
+      - name: Login to Docker Hub
+        if: steps.cfg.outputs.needs-docker-hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Configure AWS Credentials
+        if: steps.cfg.outputs.needs-ecr
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: arn:aws:iam::622024652768:role/ecr-vast-ce-github-access
+          aws-region: eu-west-1
+      - name: Login to AWS ECR
+        if: steps.cfg.outputs.needs-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+      - name: Install Nix
+        uses: cachix/install-nix-action@v21
+        with:
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup Cachix
+        uses: cachix/cachix-action@v12
+        with:
+          name: vast
+          signingKey: "${{ secrets.CACHIX_VAST_SIGNING_KEY }}"
+          pushFilter: "vast-ee"
```

Comment:
We should whitelist the open source edition here.

Reply:
Not as easy as first thought, because we'd have to whitelist all the packages we depend on as well. I came up with a better filter expression instead. The action uses `grep -vEe "$pushFilter"` internally.